packages/ui/src/components/Renderer.tsx

```diff
@@ -147,9 +147,9 @@ const Renderer = (props: RendererProps) => {
         ref.current.innerHTML = '';
       }
     };
-  }, [plugin?.ui, schema.type, reRenderDependencies]);
+  }, reRenderDependencies);
```

Comment:
the main problem was here that `reRenderDependencies` is already an array and nesting it seemed to break this